### PR TITLE
Genes for orpha disease_terms are now saved as int, matching the format for omim disease_terms

### PR DESCRIPTION
**Pull request by TereseBo**
This PR fixes a bug.
When adding disease_terms to the database terms extracted from orphadata files had the hgnc-id of linked genes saved as a string. Disease_terms extracted from omim files had the hgnc-id saved as a number. 

This PR converts the hgnc-id extracted from orphadata files to a number before saving it to a disease object

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
Image of local scout-database with orpha disease with linked gene in string format and omim disease for comparison (before change)

Screenshot 2024-01-15 at 14 55 18

Screenshot 2024-01-19 at 08 39 07


After converting the hgnc-id to int after paring the orphadata files and then updating the database:
image

File: scout/parse/orpha.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def get_orpha_to_genes_information(lines: List[str]) -> Dict[str, dict]:
 
                     if gene_source == "HGNC":
                         reference = external_reference.find("Reference").text
-                        disease["hgnc_ids"].add(reference)
+                        disease["hgnc_ids"].add(int(reference))
```
Comment (northwestwitch) [inline scout/parse/orpha.py:53]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
Looks good. I think it is fine to merge directly, since we are going to test it in the new release branch anyway!

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 - Analysis type and direct link from cases list for OGM cases
 - Removed unused `case_obj` parameter from server/blueprints/variant/controllers/observations function
 ### Fixed
+- All disease_terms have their genes in number format when added to the scout database
```
Comment (northwestwitch) [inline CHANGELOG.md:16]:
```suggestion
- All disease_terms have gene HGNC ids as integers when added to the scout database
```

**Pull request merged**